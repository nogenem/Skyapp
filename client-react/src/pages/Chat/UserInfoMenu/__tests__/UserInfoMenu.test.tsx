import React from 'react';
import { Provider } from 'react-redux';

import { fireEvent, render } from '@testing-library/react';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import type { IAppState } from '~/redux/store';
import type { IUser } from '~/redux/user/types';

import { UserInfoMenu, UnconnectedUserInfoMenu } from '../index';

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

const emptyState: Partial<IAppState> = {};
const renderWithRedux = (
  ui: React.ReactNode,
  initialState: Partial<IAppState> = emptyState,
) => {
  const store = mockStore(initialState);
  return {
    ...render(<Provider store={store}>{ui}</Provider>),
  };
};

describe('Connected UserInfoMenu', () => {
  it('renders correctly', () => {
    const user: IUser = {
      _id: '1',
      nickname: 'Test User',
      email: 'test@test.com',
      confirmed: false,
      token: '123456789',
    };

    const { container, getByText } = renderWithRedux(<UserInfoMenu />, {
      user,
    });

    expect(container).toMatchSnapshot();
    expect(getByText(user.nickname)).toBeInTheDocument();
  });
});

describe('Unconnected UserInfoMenu', () => {
  it('calls `userSignedOut` when clicking on `Sign Out` menu item', () => {
    const NICKNAME = 'Test User';
    const userSignedOut = jest.fn();

    const { getByText } = render(
      <UnconnectedUserInfoMenu
        userNickname={NICKNAME}
        userSignedOut={userSignedOut}
      />,
    );

    fireEvent.click(getByText(NICKNAME));
    fireEvent.click(getByText(/sign out/i));

    expect(userSignedOut).toHaveBeenCalledTimes(1);
  });
});
