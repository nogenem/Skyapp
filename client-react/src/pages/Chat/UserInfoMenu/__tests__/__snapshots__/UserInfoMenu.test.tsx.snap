// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Connected UserInfoMenu renders correctly 1`] = `
<div>
  <div
    class="makeStyles-container-1"
  >
    <span
      class="MuiBadge-root makeStyles-badge-4"
    >
      <div
        class="MuiAvatar-root MuiAvatar-circle MuiAvatar-colorDefault"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
          />
        </svg>
      </div>
      <span
        class="MuiBadge-badge MuiBadge-anchorOriginBottomRightCircle makeStyles-anchorOriginBottomRightCircle-8 MuiBadge-colorSecondary makeStyles-dotBadge_online-6 MuiBadge-dot makeStyles-dotBadge-5"
      />
    </span>
    <div
      class="makeStyles-textContainer-2"
    >
      <span
        class="MuiTypography-root MuiTypography-subtitle1 MuiTypography-colorInherit MuiTypography-noWrap"
        title="Test User"
      >
        Test User
      </span>
      <span
        class="MuiTypography-root MuiTypography-caption MuiTypography-colorTextSecondary MuiTypography-noWrap"
        title="Some status message..."
      >
        Some status message...
      </span>
    </div>
  </div>
</div>
`;
